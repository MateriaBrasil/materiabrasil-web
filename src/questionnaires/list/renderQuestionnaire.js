import React from 'react'
import map from 'lodash/map'
import ListItem from '@material-ui/core/ListItem'
import Typography from '@material-ui/core/Typography'
import renderQuestion from './renderQuestion'

export default props => questionnaire => {
  const { label, name, questions } = questionnaire

  return (
    <ListItem
      key={questionnaire.name}
      style={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-start',
      }}
    >
      <Typography variant="h5">{name}</Typography>
      {label &&
        <Typography variant="caption">{label}</Typography>
      }
      {map(questions, renderQuestion(props))}
    </ListItem>
  )
}
