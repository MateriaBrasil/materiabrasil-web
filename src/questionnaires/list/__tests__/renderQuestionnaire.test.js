import React from 'react'
import renderer from 'react-test-renderer'

import renderQuestionnaire from '../renderQuestionnaire'

jest.mock('../renderQuestion', () => props => <div {...props}>Question</div>)

it('render correctly', () => {
  const tree = renderer
    .create(renderQuestionnaire({ foo: 'bar' })({ questionId: 1234 }))
    .toJSON()
  expect(tree).toMatchSnapshot()
})
