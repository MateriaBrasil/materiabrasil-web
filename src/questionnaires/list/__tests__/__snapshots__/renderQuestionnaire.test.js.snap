// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<li
  className="MuiListItem-root-1 MuiListItem-default-4 MuiListItem-gutters-8"
  disabled={false}
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <h5
    className="MuiTypography-root-12 MuiTypography-h5-28"
  />
</li>
`;
